src/develop/imageop.c

```diff
@@ -1282,6 +1291,20 @@ void dt_iop_gui_update_header(dt_iop_module_t *module)
   _iop_gui_update_header(module);
 }
 
+void dt_iop_set_module_in_trouble(dt_iop_module_t *module, const gboolean state)
+{
+  gboolean previous_state = module->has_trouble;
+
+  // we don't set disabled modules in trouble, that would be annoying
+  if(module->enabled)
+    module->has_trouble = state;
+  else
+    module->has_trouble = FALSE;
+
+  if(module->has_trouble != previous_state)
+    _iop_gui_update_header(module);
```

Comment:
If you remove the conditionality here it resolves (mostly) the error where white balance module header isn't being updated when moving around the history stack. There still seems to potentially be an issue where you change the history while there is a pipe computation in progress so it's not 100% reliable but it's much better without this if statement.

Reply:
There's a similar error when you discard history with modern workflow, whereby the color calibration header isn't updated until you click on it, but I've not been able to track it down yet. When discarding history the gui_changed function is called once and for some reason it thinks that the current module is not the CAT module: `CAT_instance != current_instance` is `true`

Reply:
Updating GUI after pipe finished is supposed to be done in `_develop_ui_pipe_finished_callback()` in channelmixerrgb.c, connected to ```DT_DEBUG_CONTROL_SIGNAL_CONNECT(darktable.signals, DT_SIGNAL_DEVELOP_UI_PIPE_FINISHED,
                            G_CALLBACK(_develop_ui_pipe_finished_callback), self);```.
For some reason, that call is useless.